pkg/controllers/dashboardapi/settings/settings.go

```diff
@@ -183,3 +195,11 @@ func (s *settingsProvider) cleanupUnknownSettings(settingsMap map[string]setting
 
 	return nil
 }
+
+func (s *settingsProvider) anySettingsInstalled() (bool, error) {
```

Comment:
One small nit: this seems like it doesn't really need it's own function since it's so simple. Up to you though, feel free to resolve if you feel differently.

Reply:
I think we could keep it for its name and the helpful semantics. Otherwise, the `SetAll` method is already too big with a wall of code. This little method lifts some of the code from it.

Reply:
Up to you, I can definitely see that argument.